common-libs/did-auth-lib/README.md

```diff
@@ -1,36 +1,193 @@
-const didAuthService = new DidAuthService(options)
+#### Building of the response token(client side)
 
-// At the Server side (should recieve client did and generate for it didAuthRequest)
-const didAuthRequest = await didAuthService.createDidAuthRequestToken(audienceDid, expiresAt?)
+To create a response token it is recommended to initialize the `AffinidiDidAuthService` class with `encryptedSeed` and `encryptionKey` options on the client side in the same way as for the service side.
+Use `createDidAuthResponseToken` method to create `Did-Auth` response token. It is necessary to have a fresh request token from the service.
 
-// At the Client got didAuthRequest from server:
-const auth = await didAuthService.createDidAuthResponseToken(didAuthRequest, expiresAt?)
+```ts
+/**
+ * authDidRequestToken {String} - signed JWT request token from the service
+ * options {Object} (optional) - key value object with additional options
+ * options.maxTokenValidInMs {Number} (optional) - maximum token validity period(im milliseconds)
+ */
+const responseToken = await affinidiDidAuthService.createDidAuthResponseToken(authDidRequestToken, options)
+```
 
-// Client send this auth for Server to any endpoint which required it
-// Initial Check at the Server side
+A Response token (JWT) should be used to authenticate the user at the service side. 
 
-const verifyOptions = { environment: 'dev', accessApiKey: '...' }
+As a best practice, it is recommended to use the `LocalExpiringDidAuthResponseToken` utility class that can encapsulate a `responseToken`.
+And keep track of its expiry time according to the local time of the client. This is needed in case the local time of the client can diverge from the service time.
 
-const isVErifierd = await verifyDidAuthResponseToken(auth, verifyOptions)
+```ts
+/**
+ * tokenRequestTime {Number} - token request time, recommended to use Date.now(), could be created before calling `createDidAuthRequestToken` method
+ * responseToken {String} - response token (JWT) built on the client side
+ */
+const auth = LocalExpiringDidAuthResponseToken.initialize(tokenRequestTime, responseToken)
+// Check if token is expired.
+auth.isExpiredAt(Date.now())
 ```
+#### Validating of the response token(service side)
+
+It is recommended to implement the `Authorization` header for every endpoint and expect to receive the response token from the client side.
+Each time given response token might be validated by the service using the `verifyDidAuthResponseToken` method
+
+```ts
+/**
+ * responseToken {String} - response token (JWT) built on the client side.
+ * environment {Stirng} (required) - environment, could be `dev`, `staging` or `prod`
+ * accessApiKey {Stirng} (required) - client access api key(uuid) 
+ */
+const isValid = await affinidiDidAuthService.verifyDidAuthResponseToken(responseToken, {
+  environment: environment,
+  accessApiKey: accessApiKey,
+})
+```
+
+[Generate accessApiKey](https://apikey.affinidi.com/).
```

Comment:
@oleksiipiliugin it is production only . for dev and staging it should be a different links